### PR TITLE
Add new scan status PENDING.

### PR DESCRIPTION
**Pull request by jjnicola**
**Checklist**:

- [ ] Tests
- [ ] [CHANGELOG](https://github.com/greenbone/gvm-libs/blob/master/CHANGELOG.md) Entry

### CODE REVIEW

File: osp/osp.c [hunk 1]
```diff
@@ -572,6 +572,8 @@ osp_get_scan_status_ext (osp_connection_t *connection,
       return status;
     }
 
+  if (!strcmp (entity_attribute (child, "status"), "pending"))
+    status = OSP_SCAN_STATUS_PENDING;
   if (!strcmp (entity_attribute (child, "status"), "init"))
```
Comment (mattmundell) [inline osp/osp.c:577]:
```suggestion
  else if (!strcmp (entity_attribute (child, "status"), "init"))
```

File: osp/osp.h [hunk 2]
```diff
@@ -61,6 +61,7 @@ typedef enum
 typedef enum
 {
   OSP_SCAN_STATUS_ERROR = -1, /**< Error status. */
+  OSP_SCAN_STATUS_PENDING,    /**< Init pending. */
```
Comment (mattmundell) [inline osp/osp.h:64]: Don't know if anything stores these, but it's safer to add to the end.

For example if gvmd was storing these states as integers in the db, then the stored states would all be wrong.

File: osp/osp.h [hunk 3]
```diff
@@ -65,6 +65,8 @@ typedef enum
   OSP_SCAN_STATUS_RUNNING,    /**< Running status. */
   OSP_SCAN_STATUS_STOPPED,    /**< Stopped status. */
   OSP_SCAN_STATUS_FINISHED,   /**< Finished status. */
+  OSP_SCAN_STATUS_PENDING,    /**< Init pending. */
+
```
Comment (mattmundell) [inline osp/osp.h:69]:
```suggestion
```

**Comment by jjnicola** [comment]
Thanks @mattmundell !

**Pull request merged**